packages/cbioportal-frontend-commons/src/lib/ColumnVisibilityResolver.ts

```diff
@@ -35,8 +37,10 @@ export function resolveColumnVisibility(
 
     if (columnVisibility) {
         colVis = {
-            // if a custom columnVisibility object is provided use that one
-            ...columnVisibility,
+            // if columnVisibility object is provided (e.g. when swtiching transcripts), overwrite corresponding default visibility
+            // so if columnVisibilityByColumnDefinition contains more columns than columnVisibility
+            // the extra columns can still keep the default visibility and prevent being lost
+            ..._.merge(columnVisibilityByColumnDefinition, columnVisibility),
```

Comment:
seems slight confusing because we are already in a merging process (...).  i think we could just start with ...columnVsibiilityByColumnDefinition and that would be same thing?

Reply:
We should take this opportunity to improve the naming of the inputs of this function.  It is not at all clear what they actually are and thus why the precedence is what it is.  I realize this is not your work but if you now understand what they are, lets make better names and comments.  For example, is columnVisibilityOverride = userSelectedColumnVisibility ?  
If so, then "override" is not a great name.

Reply:
Also, i don't get what the difference is between columnVisibilityByColumnDefinition and columnVisibility.

Reply:
`columnVisiblityByColumnDefinition` is computed by processing the `columns` definition passed to the table.
According to the comments [here](https://github.com/cBioPortal/cbioportal-frontend/blob/d96636d489d0348c3c76d26c300a09452b6dfc34/src/shared/components/lazyMobXTable/LazyMobXTable.tsx#L290-L298) `columnVisibility` always refers to `props.columnVisibility` except possibly once "Reset columns" has been clicked, and `columnVisibilityOverride` keeps the state of the latest user selection.

Not sure if we really need all of these. There might be a simpler way to handle the selection.